pyanaconda/modules/payloads/payload/flatpak/flatpak_manager.py

```diff
@@ -0,0 +1,266 @@
+        self._source = None
+        self._skip_installation = False
+        self._collection_location = None
+        self._progress: Optional[ProgressReporter] = None
+        self._transaction = None
+
+    def set_sources(self, sources: List[PayloadSourceBase]):
+        """Set the source object we use to download Flatpak content.
+
+        If unset, preinstallation will install directly from the configured
+        Flatpak remote (see flatpak_remote in the anaconda configuration).
+
+        :param str url: URL pointing to the Flatpak content
+        """
+
+        source = sources[0]
+
+        if isinstance(source, RepositorySourceMixin):
+            if self._source and isinstance(self._source, FlatpakStaticSource) \
+                    and self._source.repository_config == source.repository:
+                return
+            self._source = FlatpakStaticSource(source.repository, relative_path="Flatpaks")
+        elif source.type in (SourceType.CDN, SourceType.CLOSEST_MIRROR):
+            if self._source and isinstance(self._source, FlatpakRegistrySource):
+                return
+            _, remote_url = conf.payload.flatpak_remote
+            log.debug("Using Flatpak registry source: %s", remote_url)
+            self._source = FlatpakRegistrySource(remote_url)
+        else:
+            self._source = None
+
+    def set_flatpak_refs(self, refs: Optional[List[str]]):
+        """Set the Flatpak refs to be installed.
+
+        :param refs: List of Flatpak refs to be installed, None to use
+           all Flatpak refs from the source. Each ref should be in the form
+           [<collection_id>:](app|runtime)/<id>/[<arch>]/<branch>
+        """
+        self._skip_installation = False
+        self._flatpak_refs = refs if refs is not None else []
+
+    def set_download_location(self, path: str):
+        """Sets a location that can be used for temporary download of Flatpak content.
+
+        :param path: parent directory to store downloaded Flatpak content
+           (the download should be to a subdirectory of this path)
+        """
+        self._download_location = path
+
+    @property
+    def download_location(self) -> str:
+        """Get the download location."""
+        return self._download_location
+
+    def _get_source(self):
+        if self._source is None:
+            if self._source_repository:
+                log.debug("Using Flatpak source repository at: %s/Flatpaks",
+                          self._source_repository.url)
+                self._source = FlatpakStaticSource(self._source_repository,
+                                                   relative_path="Flatpaks")
+            else:
+                _, remote_url = conf.payload.flatpak_remote
+                log.debug("Using Flatpak registry source: %s", remote_url)
+                self._source = FlatpakRegistrySource(remote_url)
+
+        return self._source
+
+    def calculate_size(self, progress: Optional[ProgressReporter]):
```

Comment:
It looks that `progress` variable is not used here.

Reply:
Yeah, it's not used. I was basically keeping things parallel with download() in case we want add progress for this operation in the future. But I can remove it.